tests/describe_task_queue_test.go

```diff
@@ -289,3 +337,125 @@ func (s *DescribeTaskQueueSuite) validateDescribeTaskQueue(
 		}, 6*time.Second, 100*time.Millisecond)
 	}
 }
+
+// validateDescribeTaskQueuePartition calls DescribeTaskQueuePartition to fetch the stats into the partition; used for testing the
+// DescribeTaskQueue caching behaviour
+func (s *DescribeTaskQueueSuite) validateDescribeTaskQueuePartition(tqName string, expectedBacklogCount map[enumspb.TaskQueueType]int64,
+	expectedAddRate map[enumspb.TaskQueueType]bool, expectedDispatchRate map[enumspb.TaskQueueType]bool) {
+	s.EventuallyWithT(func(t *assert.CollectT) {
+		resp, err := s.GetTestCluster().MatchingClient().DescribeTaskQueuePartition(
+			context.Background(),
+			&matchingservice.DescribeTaskQueuePartitionRequest{
+				NamespaceId: s.GetNamespaceID(s.Namespace()),
+				TaskQueuePartition: &taskqueuespb.TaskQueuePartition{
+					TaskQueue:     tqName,
+					TaskQueueType: enumspb.TASK_QUEUE_TYPE_WORKFLOW, // since we have only workflow tasks
```

Comment:
why are the function inputs maps keyed by `TaskQueueType` if you're hard-coding task queue type WF here? (question could be asked vice versa)

Reply:
I had done this because:

I like the idea of using maps with, activity and workflow types as keys, since we have used them in all our other tests validating the describe task queue api (for the purpose of consistency in our tests). Granted that in the caching test we are only adding workflows, I still thought it was good practice to send in the whole map rather than just the workflow type values. I can change the map in the test to just hold `enumspb.TASK_QUEUE_TYPE_WORKFLOW` if you like but I think we should keep as is

Reply:
ok that's fine with me